src/v/transform/logging/log_manager.cc

```diff
@@ -0,0 +1,294 @@
+      : _as(as)
+      , _client(c)
+      , _interval_ms(std::move(fi))
+      , _jitter(_interval_ms(), jitter_amt) {
+        _interval_ms.watch([this]() {
+            _jitter = simple_time_jitter<ClockType>{_interval_ms(), jitter_amt};
+        });
+    }
+
+    template<typename BuffersT>
+    ss::future<> start(ss::gate& gate, BuffersT* bufs) {
+        ssx::spawn_with_gate(
+          gate, [this, bufs]() -> ss::future<> { return flush_loop(bufs); });
+        return ss::now();
+    }
+
+    void wakeup() { _wakeup_signal.signal(); }
+
+    ss::future<> stop() {
+        _wakeup_signal.broken();
+        return ss::now();
+    }
+
+    ss::future<> flush_loop(auto* bufs) {
+        while (!_as->abort_requested()) {
+            try {
+                // the duration overload passes now() + dur to the timepoint
+                // overload, but the now() calculation is tied to an underlying
+                // clock type, so it doesn't work with ss::manual_clock. the
+                // timepoint overload template has a clocktype parameter, so we
+                // use that one to get the behavior we want for testing
+                if constexpr (std::is_same_v<ClockType, ss::manual_clock>) {
+                    co_await _wakeup_signal.wait(
+                      ClockType::now() + _jitter.base_duration());
+                } else {
+                    co_await _wakeup_signal.wait(_jitter());
+                }
+            } catch (const ss::broken_condition_variable&) {
+                break;
+            } catch (const ss::condition_variable_timed_out&) {
+            }
+            co_await flush(bufs);
+        }
+        co_return;
+    }
+
+    ss::future<> flush(auto* bufs) {
+        size_t n_events = 0;
+        absl::flat_hash_map<model::partition_id, io::json_batches> batches{};
+
+        if (bufs == nullptr) {
+            vlog(tlg_log.error, "Missing buffers");
+            co_return;
+        }
+
+        co_await ss::max_concurrent_for_each(
+          *bufs, MAX_CONCURRENCY, [this, &batches, &n_events](auto& pr) {
+              auto& [n, q] = pr;
+              if (q.empty()) {
+                  return ss::now();
+              }
+              // immediately swap in an empty queue. semaphore units associated
+              // with buffered events are not returned until the corresponding
+              // event is serialized.
+              auto events = std::exchange(q, {});
+
+              n_events += events.size();
+              model::transform_name_view nv{n};
+              auto pid = _client->compute_output_partition(nv);
+
+              return do_serialize_log_events(nv, std::move(events))
+                .then([name = n, pid, &batches](auto ev_json) -> ss::future<> {
+                    auto [it, _] = batches.try_emplace(pid, io::json_batches{});
+                    it->second.emplace_back(
+                      model::transform_name{std::move(name)},
+                      std::move(ev_json));
```

Comment:
Can this be simplified to `batches[pid].emplace_back(...)`?

Reply:
just sort of instinctively avoiding `operator[]` as a general rule, but ya I think it's appropriate here